lightwood/data/splitter.py

```diff
@@ -19,16 +22,27 @@ def splitter(data: pd.DataFrame, k: int, tss: TimeseriesSettings) -> List[pd.Dat
             folds = np.array_split(data, k)
         else:
             gcols = tss.group_by
-            all_group_combinations = list(product(*[data[gcol].unique() for gcol in tss.group_by]))
-            folds = [pd.DataFrame() for _ in range(k)]
-            for group in all_group_combinations:
-                subframe = data
-                for idx, gcol in enumerate(gcols):
-                    subframe = subframe[subframe[gcol] == group[idx]]
+            folds = grouped_ts_splitter(data, k, gcols)
 
-                subfolds = np.array_split(subframe, k)
+    return folds
+
+
+def grouped_ts_splitter(data: pd.DataFrame, k: int, gcols: List[str]):
+    """
+    Splitter for grouped time series tasks, where there is a set of `gcols` columns by which data is grouped.
+    Each group yields a different time series, and the splitter generates `k` folds from `data`,
+    with equally-sized sub-series for each group.
+    """
+    all_group_combinations = list(product(*[data[gcol].unique() for gcol in gcols]))
+    folds = [pd.DataFrame() for _ in range(k)]
+    for group in all_group_combinations:
+        subframe = data
```

Comment:
This might not apply: do we want to use deepcopy here as opposed to subframe=data?

Reply:
Well, these subsets should form a partition over the original dataframe, so I'd think it's not a problem.